### PR TITLE
Clic 35 sale by company report graph

### LINKED ISSUE #28: CLIC-35 Sales By Company Report graph
As a business owner, I want a report that provides illustrative charts and/or graphs for the Sales By Company Report. Visual enhancements such as these will aid in my business decision making.

TASK DESCRIPTION

By the end of the iteration, the ClickIt Sales By Company Report should have a chart and / or graph in the form view.

HOW TO EVALUATE

Go to **Custom Reports** -> **Custom Reports** (navbar) -> **Sales By Company Report**. Create a Sales By Company Report. After saving, this data should be illustrated in the form view of the Sales By Company Report.

### PR DESCRIPTION
**Pull request by kevinmurphy5**
Closes #28 

TASK OUTCOME

Task was completed as required by the corresponding issue.

As user, update **Custom Reports**. Go to Custom Reports -> **Custom Reports** (navbar) -> **Sales by Company**. Should be redirected to **Sales by Company**. Create a new Sales by Company Report by entering start date, end date, and selecting company.

On save, user should be able to view a pivot table with the sales for that company within the time range. As well as the newly added graphical representation of the data.

### CODE REVIEW

**Review by whagan — APPROVED** [review]
In a later merge, add demo data for company sales. Code works. Current merge approved.

File: custom_reports/static/src/js/sales_by_company_graph.js [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+            var canvas = this.$('canvas')[0];
+            var context = canvas.getContext('2d');
+            return new Chart(context, this.config);
+        },
+
+        /**
+         * Returns an array of colors
+         * @private
+         */
+        _getColors: function(length)    {
```
Comment (chang3d) [inline custom_reports/static/src/js/sales_by_company_graph.js:209]: This has a similar problem as #50. There will limit the number of elements you can graph to 7.

**Review by chang3d — CHANGES_REQUESTED** [review]
check the _getcolors function

**Comment by whagan** [comment]
Fixed issue on _getColors. Originally a mistake on my end. Credit @chang3d for catch and fix.

**Review by jsalajka — APPROVED** [review]
Tested and approved

**Review by jsalajka — APPROVED** [review]

**Review by jsalajka — APPROVED** [review]

**Review by chang3d — APPROVED** [review]

**Pull request merged**